flink-runtime/src/main/java/org/apache/flink/runtime/state/RegisteredBroadcastBackendStateMetaInfo.java

```diff
@@ -0,0 +1,232 @@
+				'}';
+	}
+
+	/**
+	 * A consistent snapshot of a {@link RegisteredOperatorBackendStateMetaInfo}.
+	 */
+	public static class Snapshot<K, V> {
+
+		private String name;
+		private OperatorStateHandle.Mode assignmentMode;
+		private TypeSerializer<K> keySerializer;
+		private TypeSerializer<V> valueSerializer;
+		private TypeSerializerConfigSnapshot keySerializerConfigSnapshot;
+		private TypeSerializerConfigSnapshot valueSerializerConfigSnapshot;
+
+		/** Empty constructor used when restoring the state meta info snapshot. */
+		Snapshot() {}
+
+		private Snapshot(
+				final String name,
+				final OperatorStateHandle.Mode assignmentMode,
+				final TypeSerializer<K> keySerializer,
+				final TypeSerializer<V> valueSerializer,
+				final TypeSerializerConfigSnapshot keySerializerConfigSnapshot,
+				final TypeSerializerConfigSnapshot valueSerializerConfigSnapshot) {
+
+			this.name = Preconditions.checkNotNull(name);
+			this.assignmentMode = Preconditions.checkNotNull(assignmentMode);
+			this.keySerializer = Preconditions.checkNotNull(keySerializer);
+			this.valueSerializer = Preconditions.checkNotNull(valueSerializer);
+			this.keySerializerConfigSnapshot = Preconditions.checkNotNull(keySerializerConfigSnapshot);
+			this.valueSerializerConfigSnapshot = Preconditions.checkNotNull(valueSerializerConfigSnapshot);
+		}
+
+		public String getName() {
+			return name;
+		}
+
+		void setName(String name) {
+			this.name = name;
+		}
+
+		public OperatorStateHandle.Mode getAssignmentMode() {
+			return assignmentMode;
+		}
+
+		void setAssignmentMode(OperatorStateHandle.Mode mode) {
+			this.assignmentMode = mode;
+		}
+
+		public TypeSerializer<K> getKeySerializer() {
+			return keySerializer;
+		}
+
+		void setKeySerializer(TypeSerializer<K> serializer) {
+			this.keySerializer = serializer;
+		}
+
+		public TypeSerializer<V> getValueSerializer() {
+			return valueSerializer;
+		}
+
+		void setValueSerializer(TypeSerializer<V> serializer) {
+			this.valueSerializer = serializer;
+		}
+
+		public TypeSerializerConfigSnapshot getKeySerializerConfigSnapshot() {
+			return keySerializerConfigSnapshot;
+		}
+
+		void setKeySerializerConfigSnapshot(TypeSerializerConfigSnapshot configSnapshot) {
+			this.keySerializerConfigSnapshot = configSnapshot;
+		}
+
+		public TypeSerializerConfigSnapshot getValueSerializerConfigSnapshot() {
+			return valueSerializerConfigSnapshot;
+		}
+
+		void setValueSerializerConfigSnapshot(TypeSerializerConfigSnapshot configSnapshot) {
+			this.valueSerializerConfigSnapshot = configSnapshot;
+		}
+
+		@Override
+		public boolean equals(Object obj) {
+			if (obj == this) {
+				return true;
+			}
+
+			if (!(obj instanceof RegisteredBroadcastBackendStateMetaInfo.Snapshot)) {
+				return false;
+			}
+
+			RegisteredBroadcastBackendStateMetaInfo.Snapshot snapshot =
+					(RegisteredBroadcastBackendStateMetaInfo.Snapshot) obj;
+
+			// need to check for nulls because serializer and config snapshots may be null on restore
```

Comment:
I don't think the serializer and configs will be null in this case ...

It _used_ to maybe be null in the past because previous versions did not have the config written.
For adding broadcast state now, we should be able to ensure that there is always a restored config, and for unloadable serializers, it should be replaced with a `UnloadableDummyTypeSerializer`.

Reply:
That would entail that we shouldn't be loose and use the `Objects.equals` call below.